dask/array/core.py

```diff
@@ -2883,7 +2883,24 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     --------
     stack
     """
+    seq = [asarray(a) for a in seq]
+
+    # Coerce all arrays to the same type
+    # Empty arrays can impact the output dtype
+    seq_dtypes = [a.dtype for a in seq]
+    dt = (
+        reduce(np.promote_types, seq_dtypes)
+        if len(set(seq_dtypes)) > 1 else seq_dtypes[0]
+    )
+    seq = [a.astype(dt) for a in seq]
+
+    # Drop empty arrays
+    seq = [a for a in seq if a.size]
+
     n = len(seq)
+    if n == 0:
+        return from_array(np.empty(shape=(), dtype=dt))
```

Comment:
One thing we still need to sort out is how to correctly determine the shape. Please see [this comment]( https://github.com/dask/dask/pull/4167#issuecomment-435464372 ) for details. Will give this some thought, but if you have ideas @jrbourbeau, please share. 🙂

Reply:
Have pushed a commit that should address this.